src/mapping/Mapping.cpp

```diff
@@ -130,14 +130,7 @@ bool Mapping::hasConstraint(const Constraint &constraint) const
 
 bool operator<(Mapping::MeshRequirement lhs, Mapping::MeshRequirement rhs)
 {
-  switch (lhs) {
-  case (Mapping::MeshRequirement::UNDEFINED):
-    return rhs != Mapping::MeshRequirement::UNDEFINED;
-  case (Mapping::MeshRequirement::VERTEX):
-    return rhs == Mapping::MeshRequirement::FULL;
-  case (Mapping::MeshRequirement::FULL):
-    return false;
-  };
+  return int(lhs) < int(rhs);
```

Comment:
Let's use a `static_cast<int>` here to make this more explicit.

Also, you should now enforce the underlying type of the enum to be int: 
```cpp
enum class MeshRequirement : int {
```